features/lorawan/system/lorawan_data_structures.h

```diff
@@ -0,0 +1,2896 @@
+    /*!
+     * LoRaWAN uplink counter
+     *
+     * Related MIB type: \ref MIB_UPLINK_COUNTER
+     */
+    uint32_t UpLinkCounter;
+    /*!
+     * LoRaWAN downlink counter
+     *
+     * Related MIB type: \ref MIB_DOWNLINK_COUNTER
+     */
+    uint32_t DownLinkCounter;
+    /*!
+     * Multicast channel
+     *
+     * Related MIB type: \ref MIB_MULTICAST_CHANNEL
+     */
+    MulticastParams_t* MulticastList;
+    /*!
+     * System overall timing error in milliseconds
+     *
+     * Related MIB type: \ref MIB_SYSTEM_MAX_RX_ERROR
+     */
+    uint32_t SystemMaxRxError;
+    /*!
+     * Minimum required number of symbols to detect an RX frame
+     *
+     * Related MIB type: \ref MIB_MIN_RX_SYMBOLS
+     */
+    uint8_t MinRxSymbols;
+    /*!
+     * Antenna gain
+     *
+     * Related MIB type: \ref MIB_ANTENNA_GAIN
+     */
+    float AntennaGain;
+}MibParam_t;
+
+/*!
+ * LoRaMAC MIB-RequestConfirm structure
+ */
+typedef struct eMibRequestConfirm
+{
+    /*!
+     * MIB-Request type
+     */
+    Mib_t Type;
+
+    /*!
+     * MLME-RequestConfirm parameters
+     */
+    MibParam_t Param;
+}MibRequestConfirm_t;
+
+/*!
+ * LoRaMAC TX information
+ */
+typedef struct sLoRaMacTxInfo
+{
+    /*!
+     * Defines the size of the applicable payload that can be processed.
+     */
+    uint8_t MaxPossiblePayload;
+    /*!
+     * The current payload size, dependent on the current datarate.
+     */
+    uint8_t CurrentPayloadSize;
+}LoRaMacTxInfo_t;
+
+/*!
+ * LoRaMAC status.
+ */
+typedef enum eLoRaMacStatus
+{
+    /*!
+     * Service started successfully.
+     */
+    LORAMAC_STATUS_OK,
+    /*!
+     * Service not started - LoRaMAC is busy.
+     */
+    LORAMAC_STATUS_BUSY,
+    /*!
+     * Service unknown.
+     */
+    LORAMAC_STATUS_SERVICE_UNKNOWN,
+    /*!
+     * Service not started - invalid parameter.
+     */
+    LORAMAC_STATUS_PARAMETER_INVALID,
+    /*!
+     * Service not started - invalid frequency.
+     */
+    LORAMAC_STATUS_FREQUENCY_INVALID,
+    /*!
+     * Service not started - invalid datarate.
+     */
+    LORAMAC_STATUS_DATARATE_INVALID,
+    /*!
+     * Service not started - invalid frequency and datarate.
+     */
+    LORAMAC_STATUS_FREQ_AND_DR_INVALID,
+    /*!
+     * Service not started - the device is not in a LoRaWAN.
+     */
+    LORAMAC_STATUS_NO_NETWORK_JOINED,
+    /*!
+     * Service not started - payload length error.
+     */
+    LORAMAC_STATUS_LENGTH_ERROR,
+    /*!
+     * Service not started - the device is switched off.
+     */
+    LORAMAC_STATUS_DEVICE_OFF,
+    /*!
+     * Service not started - the specified region is not supported
+     * or not activated with preprocessor definitions.
+     */
+    LORAMAC_STATUS_REGION_NOT_SUPPORTED,
+    /*!
+     * Crypto methods failure.
+     */
+    LORAMAC_STATUS_CRYPTO_FAIL,
+}LoRaMacStatus_t;
+
+/*!
+ * LoRaMAC events structure.
+ * Used to notify upper layers of MAC events.
+ */
+typedef struct sLoRaMacPrimitives
```

Comment:
how these struct are different from the rest of them below? These are using prefixes like `sSomething` plus camel cases, not consistent.

same for enum above `eLoRaMacStatus` and the others above. camelcasses in the names + members, the below these lines it looks differently, as it should. line 1840 or 1850 for instance.

Reply:
This file contains data structures used in both Semtech code and our code. Camel case data structures belong to Semtech code. Our data structures are their for translation purposes. At some point if we decide to refactor LoRaMac.cpp to make it a class instead of a monolith, we may ditch those camel case data structures and then we will have consistent code. At this time, there is no such plan.